### ISSUE TITLE
perf(model): drop Dropout from inference path (P4)

### ISSUE THREAD

**Issue by madsbolaris** [madsbolaris/MageZero#4]
**Expected speedup:** 1.3-1.5× inference latency on CPU at batch=1.

## Problem
[`src/magezero/model.py#L66`](../blob/macos-support/src/magezero/model.py#L66):
```python
self.embedding_dropout = nn.Dropout(p=0.5)
...
emb = self.embedding_dropout(emb)  # line 101
```

Although `model.eval()` disables Dropout's masking, the layer still runs an RNG check per forward call. At batch=1 on CPU this is non-trivial overhead.

## Fix
Skip the layer entirely when not training:
```python
def forward(self, indices, offsets):
    emb = self.embedding(indices, offsets)
    if self.training:
        emb = self.embedding_dropout(emb)
    ...
```

## Risk
Low. Behaviour is identical to current `eval()`-mode behaviour by design.

**Comment by madsbolaris** [comment]
Shipped on `macos-support` (commit ebaadbc). Smoke pass.
